hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/codec/KeyValueWrapper.java

```diff
@@ -0,0 +1,146 @@
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.store.contracts.precompile.codec;
+
+import com.google.protobuf.ByteString;
+import com.hedera.services.exceptions.InvalidTransactionException;
+import com.hedera.services.legacy.core.jproto.JECDSASecp256k1Key;
+import com.hedera.services.legacy.core.jproto.JEd25519Key;
+import com.hederahashgraph.api.proto.java.ContractID;
+import com.hederahashgraph.api.proto.java.Key;
+import com.hederahashgraph.api.proto.java.ResponseCodeEnum;
+
+public final class KeyValueWrapper {
+    public enum KeyValueType {
+        INVALID_KEY,
+        INHERIT_ACCOUNT_KEY,
+        CONTRACT_ID,
+        DELEGATABLE_CONTRACT_ID,
+        ED25519,
+        ECDSA_SECPK256K1
+    }
+
+    /* ---  Only 1 of these values should be set when the input is valid. --- */
+    private final boolean shouldInheritAccountKey;
+    private final ContractID contractID;
+    private final byte[] ed25519;
+    private final byte[] ecdsaSecp256k1;
+    private final ContractID delegatableContractID;
+    private final KeyValueWrapper.KeyValueType keyValueType;
+
+    /* --- This field is populated only when `shouldInheritAccountKey` is true --- */
+    private Key inheritedKey;
+
+    public KeyValueWrapper(
+            final boolean shouldInheritAccountKey,
+            final ContractID contractID,
+            final byte[] ed25519,
+            final byte[] ecdsaSecp256k1,
+            final ContractID delegatableContractID) {
+        this.shouldInheritAccountKey = shouldInheritAccountKey;
+        this.contractID = contractID;
+        this.ed25519 = ed25519;
+        this.ecdsaSecp256k1 = ecdsaSecp256k1;
+        this.delegatableContractID = delegatableContractID;
+        this.keyValueType = this.setKeyValueType();
+    }
+
+    private boolean isContractIDSet() {
+        return contractID != null;
+    }
+
+    private boolean isDelegatableContractIdSet() {
+        return delegatableContractID != null;
+    }
+
+    public boolean isShouldInheritAccountKeySet() {
+        return shouldInheritAccountKey;
+    }
+
+    private boolean isEd25519KeySet() {
+        return ed25519.length == JEd25519Key.ED25519_BYTE_LENGTH;
+    }
+
+    private boolean isEcdsaSecp256k1KeySet() {
+        return ecdsaSecp256k1.length == JECDSASecp256k1Key.ECDSA_SECP256K1_COMPRESSED_KEY_LENGTH;
+    }
+
+    public void setInheritedKey(final Key key) {
+        this.inheritedKey = key;
+    }
+
+    private KeyValueType setKeyValueType() {
+        if (isShouldInheritAccountKeySet()) {
```

Comment:
would it be quicker to add up how many keys are set and return an error if it is not equal to 1?  There would only be one place an invalid key could be returned and would make it more readable.